Document how to recover unrecovered full backup (#2601)
neo4j/docs-operations: modules/ROOT/pages/backup-restore/aggregate.adoc

----
======
=====

> [[aggregate-unrecovered-full-backup]]

Reviewer: You can aggregate an unrecovered backup in 5.26, yes, but I don't think it's possible to produce an unrecovered backup that 5.26 `neo4j-admin` is _supposed_ to work with. We don't promise that you can take a backup artefact to an older version than it was taken from and use it. So if they're taking Aura snapshots to use in self managed, their self-managed needs to be up to date with what's deployed in Aura.

=== Aggregating an unrecovered full backup

To check if a full backup is in a recovered state, run the `neo4j-admin backup inspect` command:

[source,shell]
----
bin/neo4j-admin backup inspect /mnt/backups/neo4j-2025-09-10T08-14-56.backup
----

The output will include the `RECOVERED` column.
If it shows `false`, the backup is not yet recovered:

[output]
----
|                                                 FILE | DATABASE |                          DATABASE ID |          TIME (UTC) | FULL | COMPRESSED | LOWEST TX | HIGHEST TX | STORE ID HASH | RECOVERED |
| file:///mnt/backups/neo4j-2025-09-10T08-14-56.backup |    neo4j | 4e98fab8-6ae6-46cc-84c0-61ee8a8db60a | 2025-09-10T08:14:56 | true |       true |         1 |          3 |    -116350002 |     false |
----

> If your full backup is not recovered, you have to aggregate it first.

Reviewer: I'm not keen on the 'have to' here. An unrecovered backup is still usable in most situations, such as `restore`. Perhaps:
```suggestion
If your full backup is not recovered, aggregating it will perform recovery.
```

Run the following command:

[source,shell]
----
bin/neo4j-admin backup aggregate --from-path=/mnt/backups/neo4j-2025-09-10T08-14-56.backup --keep-old-backup=true
----

This command will:

* Generate a new, recovered backup alongside the existing one.
* Preserve the original backup file (due to `--keep-old-backup=true`).

> 

Reviewer: The filename will be generated, so the database name and the current time as part of it. I have no feelings either way about whether to mention that or not.
